plugin/plugins/filesystem/nodes/root.go

```diff
@@ -0,0 +1,139 @@
+package fsnodes
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/djdv/p9/p9"
+	cid "github.com/ipfs/go-cid"
+	logging "github.com/ipfs/go-log"
+	coreiface "github.com/ipfs/interface-go-ipfs-core"
+	corepath "github.com/ipfs/interface-go-ipfs-core/path"
+	"github.com/multiformats/go-multihash"
+)
+
+var _ p9.File = (*RootIndex)(nil)
+
+func newRootPath(path string) corepath.Resolved {
+	return rootNode(path)
+}
+
+type rootNode string
+
+func (rn rootNode) String() string { return string(rn) }
+func (rootNode) Namespace() string { return nRoot }
+func (rootNode) Mutable() bool     { return true }
+func (rootNode) IsValid() error    { return nil } //TODO: should return ENotImpl
+func (rn rootNode) Cid() cid.Cid {
+	prefix := cid.V1Builder{Codec: cid.DagCBOR, MhType: multihash.BLAKE2B_MIN}
+	c, err := prefix.Sum([]byte(rn))
+	if err != nil {
+		panic(err) //invalid root
+	}
+	return c
+}
+func (rootNode) Root() cid.Cid { //TODO: this should probably reference a package variable set during init `rootCid`
```

Comment:
The root shouldn't _have_ a CID.

Reply:
We now return an empty/undefined one when this is called